src/psyclone/tests/domain/nemo/transformations/enter_data_test.py

```diff
@@ -0,0 +1,94 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: R. W. Ford, A. R. Porter, S. Siso and N. Nobre, STFC Daresbury Lab
+
+'''Module containing py.test tests for the transformation of the PSy
+   representation of NEMO code using the OpenACC enterdata directive.
+
+'''
+
+import os
+
+from fparser.common.readfortran import FortranStringReader
+from psyclone.psyGen import PSyFactory
+from psyclone.transformations import ACCEnterDataTrans, ACCKernelsTrans
+
+
+# Constants
+API = "nemo"
+# Location of the Fortran files associated with these tests
+BASE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "../../test_files")
+
+# Test code with explicit NEMO-style do loop
+EXPLICIT_DO = ("program explicit_do\n"
+               "  REAL :: r\n"
+               "  INTEGER :: ji, jj, jk\n"
+               "  INTEGER, PARAMETER :: jpi=3, jpj=5, jpk=7\n"
+               "  REAL, DIMENSION(jpi, jpj, jpk) :: umask\n"
+               "  DO jk = 1, jpk\n"
+               "     DO jj = 1, jpj\n"
+               "        DO ji = 1, jpi\n"
+               "           umask(ji,jj,jk) = ji*jj*jk/r\n"
+               "        END DO\n"
+               "     END DO\n"
+               "  END DO\n"
+               "end program explicit_do\n")
+
+
+def test_explicit(parser):
+    '''
+    Check code generation for enclosing a single explicit loop containing a
+    kernel inside a data region.
+
+    '''
+    reader = FortranStringReader(EXPLICIT_DO)
+    code = parser(reader)
+    psy = PSyFactory(API, distributed_memory=False).create(code)
+    schedule = psy.invokes.get('explicit_do').schedule
+    acc_kernels = ACCKernelsTrans()
+    acc_kernels.apply(schedule.children)
+    acc_trans = ACCEnterDataTrans()
+    acc_trans.apply(schedule)
+    gen_code = str(psy.gen).lower()
```

Comment:
Please use `code = fortran_writer(schedule)` where fortran_writer is a fixture.
psy.gen is going to be deleted at some point and nemo uses the Writer anyway